mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -523,6 +524,48 @@ class GetRangeSizePattern : public ConvertOpToLLVMPattern<Op> {
   }
 };
 
+/// Helper function to calculate the linear ID using ID and range getters.
+template <typename IDGetter, typename RangeGetter>
+static void getLinearIDRewriter(Operation *op, unsigned dimension,
+                                IDGetter getID, RangeGetter getRange,
+                                ConversionPatternRewriter &rewriter) {
+  const auto loc = op->getLoc();
+  Value newValue;
+  switch (dimension) {
+  case 1:
```

Comment:
could we do something like...
```
assert(dimension >= 1 && dimension <= 3 && ...);
for (unsigned i = 0; i < dimension; ++i) {
  newValue = newValue ? rewriter.create<arith::AddIOp>(newValue, getID(i)) : getID(i);
  for (unsigned j = i+1; j < dimension; ++j)
    newValue = rewriter.create<arith::MulIOp>(newValue, getRange(j));
}
```

Reply:
I modified your version a bit to avoid changing the tests and adding duplicated instructions.